gst/nnstreamer/nnstreamer_plugin_api_filter.h

```diff
@@ -184,41 +264,93 @@ typedef struct _GstTensorFilterFramework
+       * @return 0 if OK. non-zero if error.
+       *
+       * @note CAUTION: private_data can be NULL if the framework is not yet opened by the caller.
        */
 
-  int (*reloadModel) (const GstTensorFilterProperties * prop, void **private_data);
-      /**< Optional. tensor_filter.c will call it when a model property is newly configured. Also, 'is-updatable' property of the framework should be TRUE. This function reloads a new model specified in the 'prop' argument. Note that it requires extra memory size enough to temporarily hold both old and new models during this function to hide the reload overhead.
+      int (*getModelInfo) (const GstTensorFilterProperties * prop,
+          void *private_data, model_info_ops ops, GstTensorsInfo *in_info,
+          GstTensorsInfo *out_info);
+      /**< Mandatory callback. Gets the model related tensor info.
+       * If ops == GET_INOUT_INFO, in_info would contain the input tensor info, and out_info would contain the output tensor info.
+       * If ops == SET_INPUT_INFO, in_info would contain the provided input tensor info, and out_info would contain the updated output tensor info.
+       * At least one of SET_INPUT_INFO and GET_INOUT_INFO operations must be supported.
+       *
+       * Note: Tensor_Filter::main will configure input dimension from pad-cap in
+       * run-time for the sub-plugin. Then, the sub-plugin is required to return
+       * corresponding output dimension, and will call SET_INPUT_INFO operation.
+       *
+       * Note: With SET_INPUT_INFO operation, the caller must NOT allocate or fix
+       * internal data structure based on the return value until invoke is called.
+       * Gstreamer may try different dimensions before settling down.
        *
        * @param[in] prop read-only property values
-       * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, close() frees private_data and set NULL.
-       * @return 0 if ok. < 0 if error.
+       * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer.
+       * @param[in] ops operation to be performed
+       * @param[in/out] in_info structure of input tensor info
+       * @param[out] out_info structure of output tensor info (return value)
+       * @return 0 if OK. non-zero if error. -ENOENT is operation is not supported.
        */
 
-  int (*checkAvailability) (accl_hw hw);
-      /**< Optional. Check if the provided hardware accelerator is supported. This check is static or dynamic based on framework support. Positive response of this check does not guarantee successful running of model with this accelerator. The static check can be performed without opening the framework.
+      int (*eventHandler) (const GstTensorFilterProperties * prop, void *private_data,
```

Comment:
Discussion:
Is it necessary to pass any parameter? (e.g. define new event to change accel : CHANGE_ACCEL / GPU)

Reply:
the parameter to be passed via `void *data` is optional (can be NULL).
The handler event will interpret and use `void *data` as per its need and implementation. 
In the example of change accel, `void *data` will have to interpreted as `accl_hw` or `char *`.
As additional operations should directly be able to add into `enum event_ops`, its `eventHandler` is kept generic.

Updated description of data to allow NULL.

Reply:
void * should be enough for the basic need.

However, we may be able to define a standard struct for each event type as a union of multiple struct per event type. The data structure for each event is not determined by sub-plugins; it's determined by tensor_filter and tensor_filter calls eventHandler.

Reply:
Right, it is determined by the caller and not by the callee.
Added standard struct for each event type.